raft/src/test/java/org/apache/kafka/raft/RaftClientTestContext.java

```diff
@@ -1159,18 +1164,20 @@ void readBatch(BatchReader<String> reader) {
         }
 
         @Override
-        public void handleLeaderChange(LeaderAndEpoch leader) {
+        public void handleLeaderChange(LeaderAndEpoch leaderAndEpoch) {
             // We record the next expected offset as the claimed epoch's start
-            // offset. This is useful to verify that the `handleClaim` callback
+            // offset. This is useful to verify that the `handleLeaderChange` callback
             // was not received early.
-            if (localId.isPresent() && leader.isLeader(localId.getAsInt())) {
+            if (localId.isPresent() && leaderAndEpoch.isLeader(localId.getAsInt())) {
                 long claimedEpochStartOffset = lastCommitOffset().isPresent() ?
                     lastCommitOffset().getAsLong() + 1 : 0L;
-                this.currentClaimedEpoch = OptionalInt.of(leader.epoch());
-                this.claimedEpochStartOffsets.put(leader.epoch(), claimedEpochStartOffset);
+                this.currentClaimedEpoch = OptionalInt.of(leaderAndEpoch.epoch());
```

Comment:
Since we are storing `currentLeaderAndEpoch`, I think that we can remove the variable `currentClaimedEpoch` and add a function that returns this epoch based on the value of `currentLeaderAndEpoch`. What do you think?

Reply:
Yes, makes sense.